src/tools/compiletest/src/header.rs

```diff
@@ -650,7 +646,9 @@ fn iter_header<R: Read>(testfile: &Path, cfg: Option<&str>, rdr: R, it: &mut dyn
         let ln = ln.trim();
         if ln.starts_with("fn") || ln.starts_with("mod") {
             return;
-        } else if ln.starts_with(&comment_with_brace) {
+        } else if ln.starts_with(comment)
+            && ln.trim_start_matches(comment).trim_start().starts_with('[')
```

Comment:
This is not quite right - it'll match `//// [` too. Couldn't we do `ln[comment.len()..]`?

Reply:
Ah, you're right, I couldn't consider such a case. Fixed condition.